crates/orderbook/src/api/put_app_data.rs

```diff
@@ -10,19 +10,22 @@ use {
 
 fn request(
     max_size: usize,
-) -> impl Filter<Extract = (AppDataHash, AppDataDocument), Error = Rejection> + Clone {
-    warp::path!("v1" / "app_data" / AppDataHash)
+) -> impl Filter<Extract = (Option<AppDataHash>, AppDataDocument), Error = Rejection> + Clone {
+    let opt = warp::path::param::<AppDataHash>()
```

Comment:
Optionally fetching the appdata hash looks correct but would be nice to add a test.